### DISCUSSION TITLE
Easy way to update manifest templates

### DISCUSSION THREAD

**Discussion by brunnels** [PyratLabs/ansible-role-k3s#95] [Q&A]
I'm using `k3s_server_manifests_templates` to set customized templates.  What is the best way for me to make changes to these templates and then re-deploy them via ansible?

I've also been having a problem with dns not resolving when doing the HA cluster example with the only change being:
```
    k3s_server:
      disable:
        - servicelb
        - traefik
```

I'm re-imaging all my hosts now to see if maybe it's some leftover config causing it.

**Comment by xanmanning** [comment]
Re-running the playbook used for deployment will update the files in `k3s_server_manifests_templates` if they differ on the target (converge the config).

Files deployed to the `/var/lib/rancher/k3s/server/manifests` directory should behave as described in this article: https://rancher.com/docs/k3s/latest/en/advanced/#auto-deploying-manifests

This Ansible role is only responsible for putting the file in the right place, if k3s isn't picking up changes to these files I'd suggest an issue be created with the k3s-io team :slightly_smiling_face:

**Reply by brunnels** [reply]
Okay, that is what I was doing but it doesn't seem to remove manifests that were added.  For instance I added a nginx.yaml which isn't one of the original manifests but it doesn't get removed.

Also seems updating a helm manifest doesn't remove from helm automatically according to this.  https://github.com/k3s-io/k3s/issues/737.  It says fixed but I think that was only for the service removal problem when deleting helmcharts.
